pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,207 @@
+---
+lang: en
+title: 'Decorators'
+keywords: LoopBack 4.0, LoopBack-Next
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Decorators.html
+summary:
+---
+
+A decorator allows you to annotate or modify your class declarations and members with metadata.
+
+## Introduction
+
+*If you're new to Decorators in TypeScript, see [here](https://www.typescriptlang.org/docs/handbook/decorators.html) for more info.*
+
+Decorators give LoopBack the flexibility to modify your plain TypeScript classes 
+and properties in a way that allows the framework to better understand how to 
+make use of them, without the need to inherit base classes or add functions 
+that tie into an API.
+
+Currently, LoopBack core provides built-in decorator functions you can use directly
+by `@myDecorator`:
+
+- [route](#route-decorators)
+- [inject](#dependency-injection)
+
+As a default, LoopBack also comes with some pre-defined component decorators:
+
+- [authentication](#authentication-decorators)
+- [repository](#repository-decorators)
+
+## Route Decorators
+- `@get`, `@post`, `@put`, `@patch`, `@del`
+
+  Register route for a controller by REST decorator, for their usage, refer to 
+  [Routing to Controllers](controllers.htm#Routing-to-Controllers)
+
+- `@api`
+
+  Register route for a controller by api decorator, more details and example refer 
+  to [Specifying Controller APIs](controllers.htm#Specifying-Controller-APIs)
+
+- `@operation`
+
+  Expose a Controller method as a REST API operation. Usage refer to [TBD in Controllers](controllers.htm#)
+
+- `@param`
```

Comment:
`@param` has another set of operations that make it even easier to use that I think might be worthwhile documenting. 

It follows this pattern to make it quicker to define params: `@param.${in}.${type}(${name})`
in - one of: query, header, path, formData, body
type - one of: string, number, boolean, integer
name - name of the parameter

So an example would be `@param.query.number('offset')` or `@param.query.number('pageSize')`